server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailQuery.scala

```diff
@@ -35,6 +36,33 @@ object Limit {
 }
 case class QueryState(value: String) extends AnyVal
 
+sealed trait SortProperty {
+  def toSortClause: SortClause
+}
+case class ReceivedAtSortProperty() extends SortProperty {
+  override def toSortClause: SortClause = SortClause.Arrival
+}
+
+object IsAscending {
+  val default: IsAscending = IsAscending(false)
+}
+case class IsAscending(value: Boolean) extends AnyVal {
+  def toSortOrder: SearchQuery.Sort.Order = if (value) SearchQuery.Sort.Order.NATURAL else SearchQuery.Sort.Order.REVERSE
+
+}
+
+object Comparator {
+  val default: Comparator = Comparator(ReceivedAtSortProperty(), Some(IsAscending.default), None)
+}
+
+case class Collation(value: String) extends AnyVal
+
+case class Comparator(property: SortProperty,
+                      isAscending: Option[IsAscending],
+                      collation: Option[Collation]) {
+  def toSort: SearchQuery.Sort = new SearchQuery.Sort(property.toSortClause, isAscending.get.toSortOrder)
```

Comment:
don't use 'get' on Option

Reply:
should explain your NullPointerException on reads

Reply:
Use getOrElse